File: resources/src/main/java/com/andyanika/resources/ResourceImpl.java
package com.andyanika.resources;

import android.content.Context;

import com.andyanika.translator.common.interfaces.Resources;

import javax.inject.Inject;
import javax.inject.Named;

class ResourceImpl implements Resources {
    private Context activityContext;

    @Inject
    ResourceImpl(@Named("activity") Context activityContext) {
        this.activityContext = activityContext;
    }

    @Override
    public String getString(int resourceId) {
        return activityContext.getString(resourceId);
    }

    @Override
    public String getString(String resourceName) {
        int id = activityContext.getResources().getIdentifier(resourceName, "string", activityContext.getPackageName());
        return getString(id);
    }
}

File: resources/src/main/java/com/andyanika/resources/ResourceImpl.kt
package com.andyanika.resources

import android.content.Context
import com.andyanika.translator.common.interfaces.Resources
import javax.inject.Inject
import javax.inject.Named

class ResourceImpl @Inject constructor(@Named("activity") private val activityContext: Context) :
    Resources {
    override fun getString(resourceId: Int): String {
        return activityContext.getString(resourceId)
    }

    override fun getString(resourceName: String): String {
        val id = activityContext.resources.getIdentifier(resourceName, "string", activityContext.packageName)
        return getString(id)
    }
}
